src/Compilers/CSharp/Test/Syntax/Parsing/ParsingTests.cs

```diff
@@ -254,35 +300,80 @@ protected void EOF()
             }
         }
 
-        private IEnumerable<SyntaxNodeOrToken> EnumerateNodes(CSharpSyntaxNode node, bool dump)
+        private IEnumerable<SyntaxNodeOrTokenOrTrivia> EnumerateNodes(CSharpSyntaxNode node, bool dump)
         {
             Print(node, dump);
             yield return node;
 
-            var stack = new Stack<ChildSyntaxList.Enumerator>(24);
-            stack.Push(node.ChildNodesAndTokens().GetEnumerator());
+            var stack = new Stack<(SyntaxTriviaList.Enumerator, ChildSyntaxList.Enumerator, SyntaxTriviaList.Enumerator)>(24);
+            stack.Push((default, node.ChildNodesAndTokens().GetEnumerator(), default));
             Open(dump);
 
             while (stack.Count > 0)
             {
-                var en = stack.Pop();
-                if (!en.MoveNext())
+                var (en1, en2, en3) = stack.Pop();
+
+                byte en;
+                SyntaxTrivia currentTrivia = default;
+                SyntaxNodeOrToken currentChild = default;
+
+                if (en1.MoveNext())
+                {
+                    currentTrivia = en1.Current;
+                    en = 1;
```

Comment:
I find this a little bit clunky, but, it's test code, and I don't have any ideas which are simple and obviously better to do instead.